### DISCUSSION TITLE
Using Model-Viewer in Angular for Interactive 3D Model Application with Camera and Voice Features

### DISCUSSION THREAD

**Discussion by Akhil-coditas** [google/model-viewer#4282] [General]
I have requirement for an application in Angular.
The requirements are :

1.  i need to have first page which has toggle for camera and voice permissions and language options.
2. then a second page is displayed with instructions and getstarted button.
2. Later on  clicking on getstarted button as animation of door being opened is shown (it is also a video) and then my 3d model is displayed on camera which also has voice
3. i have options at the bottom to record voice along with text to speech, play , pause 3d model and then i have other options where i need to display different models according to option clicked.

also each page has custom ui depending on application

Is model-viewer suitable for this in angular? or do i need to use some other package ?

**Comment by elalish** [comment]
I think that's doable, but it'll be some work. Only go down this route if you're quite comfortable with JS.
